File: Cow.php
<?php
spl_autoload_register(function ($class_name) {
    include $class_name . '.php';
});

class Cow extends Animal{

    public $animalId;
    public $animalBreed = 'cows';    
    public $product = 'milk';

    function __construct() {
        
        $this -> animalId = parent::$id++;
        
    }

    
    public function getOutputProduct(){

        return rand(8, 12);

    }
}

File: go/animals/Cow.go
package animals

import "math/rand"

type Cow struct {
	AnimalBreed, Product string
}

func (cow *Cow) Init() Cow {
	return Cow{
		AnimalBreed: "cows",
		Product:     "milk",
	}
}

// Returns a random amount of products.
func (cow *Cow) GetOutputProduct() int {
	return rand.Intn(5) + 8
}
